src/hooks/coreSDK/safeCoreSDK.ts

```diff
@@ -44,16 +44,37 @@ export const initSafeSDK = async (
   chainId: string,
   safeAddress: string,
   safeVersion: string,
+  chainInfo: ChainInfo,
+  walletChainId: string,
 ): Promise<Safe> => {
   let isL1SafeMasterCopy = chainId === chains.eth
   // Legacy Safe contracts
   if (isLegacyVersion(safeVersion)) {
     isL1SafeMasterCopy = true
   }
 
-  const ethersProvider = new Web3Provider(provider)
+  const signerProvider = new Web3Provider(provider)
+
+  // Monkey patch the signerProvider to proxy requests to the "readonly" provider if on the wrong chain
+  if (walletChainId !== chainId) {
+    const signerMethods = ['eth_sign', 'eth_signTypedData', 'eth_accounts', 'personal_sign']
```

Comment:
I think this is missing one [signing method](https://docs.metamask.io/guide/signing-data.html#signtypeddata-v4)

```suggestion
    const signerMethods = ['eth_sign', 'eth_signTypedData', 'eth_signTypedData_v4', 'eth_accounts', 'personal_sign']
```

Reply:
Nit: can we create an enum of these? These are used in `_tryOffChainSigning` as well. Maybe it could be a "global" enum.